### PR TITLE
81 update edit delete button - Micheale Andemariam

### LINKED ISSUE #81: Update-edit-delete-button
## Story
As a user I can edit My availability 
As a user I can delete My availability

## Acceptance Criteria

When I see my availability, I click edit and delete button. That goes to edit /delete them.
I can see success screen when finished.

## Implementation Notes

1. Update Edit and delete components

## Checklist:

- [ Y ] I have defined the user story
- [ Y ] I have defined some acceptance criteria
- [ Y ] I have included any implementation notes
- [ Y ] I have included all relevant links

### PR DESCRIPTION
**Pull request by micheale2021**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #81

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Comment by micheale2021** [comment]
Please uncomment edit and delete my availability components in MyAvailabilty.js

File: client/src/components/EditMyAvailability.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import { Form, Modal } from "react-bootstrap";
+import Button from "react-bootstrap/Button";
+function EditMyAvailability(props) {
```
Comment (RainbowGiantSquid) [inline client/src/components/EditMyAvailability.js:5]: would be good to add a line between the imports and the main component

File: client/src/components/EditMyAvailability.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  const [edit, setEdit] = useState(false);
+  const [trainees_id, setTrainees_id] = useState(localStorage.getItem("id"));
+  const [date, setDate] = React.useState(new Date());
+  const [topic, setTopic] = React.useState("");
+  const handleUpdate = (e) => {
```
Comment (RainbowGiantSquid) [inline client/src/components/EditMyAvailability.js:12]: also here, an empty line below the hooks for redability

File: client/src/components/EditMyAvailability.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        Authorization: localStorage.getItem("token"),
+      },
+      body: JSON.stringify(available),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.error) {
+          alert(data.error);
+        } else {
+          alert("You have updated this available date!");
```
Comment (RainbowGiantSquid) [inline client/src/components/EditMyAvailability.js]: could we please not use window alert? Could this be a banner component that displays this message on top of the page (for example on timeout something like 5 seconds?)

File: client/src/components/EditMyAvailability.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                  new Date(new Date().setMonth(new Date().getMonth() + 1))
+                    .toISOString()
+                    .split("T")[0]
+                }
+                name="date"
+                onChange={handleChange}
+              />
+            </Form.Group>
+            <Form.Group controlId="formBasicTopic">
+              <Form.Label style={{ color: "goldenrod" }}>Topic</Form.Label>
```
Comment (RainbowGiantSquid) [inline client/src/components/EditMyAvailability.js]: could we extract styling to a separate css file please and make it more aligned with the design? I'm worried that 'goldenrod' labels won't be accessible in terms of the contrast ratio.

**Review by RainbowGiantSquid — CHANGES_REQUESTED** [review]
Thanks - good work, some comments about similar things as previous PRs.

File: client/src/components/EditMyAvailability.js (hunk 1 above)
Comment (micheale2021) [reply client/src/components/EditMyAvailability.js:5]: Thank you, I have just fixed it.

File: client/src/components/EditMyAvailability.js (hunk 2 above)
Comment (micheale2021) [reply client/src/components/EditMyAvailability.js:12]: Thank you, I have just fixed it.

File: client/src/components/EditMyAvailability.js (hunk 4 above)
Comment (micheale2021) [reply client/src/components/EditMyAvailability.js]: I agree, for now, I have just removed the colour goldenrod. I think we will need a separate ticket to style the whole form.

File: client/src/components/EditMyAvailability.js (hunk 3 above)
Comment (micheale2021) [reply client/src/components/EditMyAvailability.js]: I created errorMessage useState for this. I can't see what the page is doing at the moment but I hope that solves the problem.

**Review by RainbowGiantSquid — APPROVED** [review]

**Pull request merged**